pkg/controller/clusterstorage/clusterstorage_controller.go

```diff
@@ -145,6 +152,15 @@ func (r *ReconcileClusterStorage) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
+	// Run validation checks
+	err = validation.CheckAlphaSnapshot(r.client)
```

Comment:
How often is `Reconcile` called? In other words, how long does it take for `Upgradeable=false` to get away after user removes v1alpha1 CRD?

Reply:
This happens fairly quickly - we requeue the request when throwing an error, and on line 161 we return an error, so it continues trying. From testing it takes seconds to clear the flag after the CRD is removed.

The other way (flagging the cluster as Upgradeable=false) takes a bit longer, as we wait for a new Reconcile request to come in and proceed through this logic. From testing it seems to be about 15 minutes.

Reply:
Performing additional tests show that even though the Reconcile request is retried fairly frequently. In most cases it's only a couple of seconds, as the request is immediately re-queued; however, it may take the full amount of time for a subsequent Reconcile request. So up to 15 minutes in worst case scenario.